src/main/java/hudson/plugins/git/extensions/impl/CheckoutOption.java

```diff
@@ -32,25 +32,72 @@ public Integer getTimeout() {
         return timeout;
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public boolean requiresWorkspaceForPolling() {
         return false;
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public void decorateCheckoutCommand(GitSCM scm, Run<?, ?> build, GitClient git, TaskListener listener, CheckoutCommand cmd) throws IOException, InterruptedException, GitException {
         cmd.timeout(timeout);
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     @Deprecated
     public void decorateCheckoutCommand(GitSCM scm, AbstractBuild<?, ?> build, GitClient git, BuildListener listener, CheckoutCommand cmd) throws IOException, InterruptedException, GitException {
         cmd.timeout(timeout);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        CheckoutOption that = (CheckoutOption) o;
+
+        return timeout != null ? timeout.equals(that.timeout) : that.timeout == null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int hashCode() {
+        return 0;
```

Comment:
the fields are all mutable, thus the hashcode must be constant... should be `CheckoutOption.class.hashCode()` if I am being fair to any hashsets/maps